tools/vulnerability-scan/wss-scan.sh

```diff
@@ -47,33 +49,67 @@ basepath=$baseDirPath"/repos"
 
 echo "Cleaning up scan directories if already present"
 rm -rf $basepath
+echo "Cleaning up temp file that will affect scan"
+rm -rf /tmp/ws*
 
 mkdir -p $basepath
  
 # clone the desired Repos for scanning 
 for repo in $gitRepos
 do
   echo "Cloning repo "$gitBasePath$repo
-  git clone "$gitBasePath$repo".git $basepath"/"$repo
+  if [ ${repo} == "OpenSearch" ]
+  then
+    echo "Cloning "$repo" at branch 1.x"
+    git clone -b 1.x "$gitBasePath$repo".git ${basepath}/${repo}
+    # /qa/ in OpenSearch repo contains all files related to the previous version of ODFE.
+    # WhiteSource will attept to download them and take hours to build or timeout error. 
+    # Remove /qa/ because it's irrelevant to OpenSearch. 
+    rm -rf ${basepath}/${repo}/qa
+  else
+    echo "Cloning "$repo" at main branch" 
+    git clone "$gitBasePath$repo".git ${basepath}/${repo}
+  fi
 done
 
 echo -n > info.txt
 target='OpenSearch'
-mvn -v; node -v; npm -v; yarn -v
 
 # scan the Repos using the WhiteSource Unified Agent
 for repo in $gitRepos
 do
   repo_path=$basepath"/"$repo
   if [ -d "$repo_path" ]
   then
+    cd $repo_path
     echo "Scanning repo: "$gitBasePath$repo " Project: " $repo
-    java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product "$target" -project $repo
+    if [ -e "build.gradle" ]
+    then
+        echo "build.gradle for $repo exist in root"
+        if [ ! -e "settings.gradle" ]
+        then
+            echo "settings.gradle not exist in $repo, create one based on build.gradle name"
+            generate_settings_gradle
+        else
+            echo "settings.gradle exist $repo"
+        fi
+    else
+        # Exceptions for some repos that have `build.gradle` in sub-folder
+        if [ ${repo} == "observability" ]
+        then
+          cd $repo_path/opensearch-observability
+          generate_settings_gradle
+        else
+          echo "build.gradle for $repo not exist, either dashboards repo or missing necessary files"
+        fi
+    fi
+    java -jar $baseDirPath/wss-unified-agent.jar -c $baseDirPath/wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product "$target" -project $repo
+    cd $baseDirPath && pwd
```

Comment:
this `cd` too

Reply:
This is the absolute path. https://github.com/opensearch-project/opensearch-build/blob/ac208d0b7a403a3782d3654fecbe7fe91d70e7d9/tools/vulnerability-scan/wss-scan.config#L1

Reply:
yes, but if we are using absolute path everywhere, then we can remove all the `cd` statements right?

Reply:
Like I said above, we have to go into each folder to run the scan. Also refer to the second point Peter listed above (https://github.com/opensearch-project/opensearch-build/pull/1411#issuecomment-1000945335)
**The scan needs to run in the root folder of each repo**